packages/core/src/types/authMachine.ts

```diff
@@ -63,7 +89,7 @@ export type AuthInputAttributes = Record<AuthInputNames, InputAttributes>;
 
 export interface AuthEvent {
   type: AuthEventTypes;
-  data?: any; // TODO: strongly type data for each AuthEventType
+  data?: any;
```

Comment:
I think we can avoid using `any`. I know we don't necessarily know everything that `data` could be, since it encompasses our form submissions, api return values, etc. But it will always at least be an "object", right? Like, `data` won't ever be `'a string'`, or a `number`. What do you think about about making it have a type of `Record<string, any>`? It still uses `any`, which I guess is sort of unavoidable in this case, but it is more specific at least.

Reply:
Right, let me do that. Eventually, we can have even stricter types with type guards and what not. For now, let me put `Record<PropertyKey, any>`.

Reply:
https://github.com/aws-amplify/amplify-ui/pull/180/commits/aa50ba3a4c14dfb0f9d5acf3227fed5298be910f